foursight_core/app_utils.py

```diff
@@ -398,7 +780,16 @@ def view_run_action(self, environ, action, params, context="/"):
         return Response(status_code=302, body=json.dumps(resp_headers),
                         headers=resp_headers)
 
-    def view_foursight(self, environ, is_admin=False, domain="", context="/"):
+    def get_unique_annotated_environment_names(self):
+        unique_environment_names = self.environment.list_unique_environment_names()
+        unique_annotated_environment_names = [
+            {"name": env,
+             "short": short_env_name(env),
+             "full": full_env_name(env),
+             "inferred": infer_foursight_from_env(envname=env)} for env in unique_environment_names]
```

Comment:
If it doesn't require too much other changing of things, I would call the fourth entry "foursight" rather than "inferred". It gets inferred, but it is a foursight name. If it's likely to break things, I'd add a TODO.